### PR TITLE
test(checkpoint): fuzz resume equivalence with spikes in flight (LIM-1223)

### PR DESCRIPTION
**Pull request by rmems**
## Summary

A valid restored `SynapticMesh` must continue **exactly like** the uninterrupted live mesh, including when spikes are still in the delay buffer.

This adds `tests/checkpoint_resume/`: a bounded, seeded property suite that generates graphs, delays, signed weights, checkpoint locations, and suffix event sequences, then compares live vs restored copies tick-for-tick.

Malformed checkpoints stay covered by the existing load-path tests (LIM-1106 / LIM-1151). This suite only serializes valid live state and never normalizes invalid payloads.

## What is compared

After the prefix and after every suffix tick:

- returned synaptic currents
- `tick`
- queued delay-buffer deliveries (checkpoint `delay_buffer` slots)
- the rest of the checkpoint snapshot (graph + buffer metadata)

Restore is exercised through **JSON** and **postcard** (dev-only dependency; the published crate still depends solely on `serde`). `bincode` 1.x is unmaintained (RUSTSEC-2025-0141), so it is not used.

## Coverage

- **CI:** 512 seeded cases plus named boundary fixtures (empty graph, single neuron, delay 0, delay capacity / headroom, multi-spike same slot, checkpoint immediately before delivery, signed weights, empty ticks).
- **Regressions:** `REGRESSION_SEEDS` for failures outside the CI range, the original delay-2 in-flight case, and a frozen JSON checkpoint captured before delivery. On failure the harness shrinks the scenario and prints a minimized counterexample to persist.
- **Nightly / ignored:** `resume_equivalence_nightly` (default 10_000 additional seeds). Documented in the test module and `REVIEW.md`:

```bash
cargo test --locked --test checkpoint_resume resume_equivalence_nightly -- --ignored
CHECKPOINT_RESUME_CASES=50000 cargo test --locked --test checkpoint_resume resume_equivalence_nightly -- --ignored
```

## Also in this PR

- Docker rustdoc verify now checks `synaptic_wiring/` after the crate rename.
- The suite is split across `main.rs` / `harness.rs` / `generate.rs` so Codacy's file-length gate stays green.

## Test plan

- [x] `cargo fmt --check`
- [x] `cargo clippy --locked --all-targets --all-features -- -D warnings`
- [x] `cargo test --locked --all-features`

Linear: [LIM-1223](https://linear.app/rpd-34/issue/LIM-1223/testcheckpoint-fuzz-resume-equivalence-with-spikes-in-flight)

Linear Issue: [LIM-1223](https://linear.app/rpd-34/issue/LIM-1223/testcheckpoint-fuzz-resume-equivalence-with-spikes-in-flight)

<div><a href="https://cursor.com/agents/bc-a2c869db-9bc3-4a2a-a74d-88b5f7c25598?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-a2c869db-9bc3-4a2a-a74d-88b5f7c25598&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

---
## Summary by cubic
Adds a seeded property suite proving that a restored `SynapticMesh` continues tick-for-tick identically to the uninterrupted live mesh, including when spikes are still in the delay buffer (LIM-1223). Also fixes the Docker rustdoc path after the crate rename to `synaptic-wiring`.

- Compares returned currents, `tick`, queued delay-buffer deliveries, and the checkpoint snapshot after every suffix tick; mismatches report differing fields and full snapshot context.
- Recipe selection avoids 32-bit truncation of seeds before modulo.
- Restore is exercised through JSON and dev-only `postcard`, so the published crate still depends solely on `serde`.
- CI runs 512 seeded cases plus named boundary and regression fixtures; an ignored nightly run is controlled by `CHECKPOINT_RESUME_CASES`.
- Invalid checkpoints are not normalized here; malformed-state rejection stays covered by the existing load-path tests (LIM-1106/LIM-1151).
- Docker verification now checks `synaptic_wiring` rustdoc to match the renamed crate; the suite is split across modules in `tests/checkpoint_resume/` to stay under Codacy's line budget.

<sup>Written for commit 4afd029fe472ee8fc509cd4756f4d9a05b90ac13. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Limen-Neural/synaptic-wiring/pull/85?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

___

## **CodeAnt-AI Description**
Verify checkpointed meshes resume exactly like uninterrupted meshes

### What Changed
- Added seeded checkpoint-resume tests covering 512 CI scenarios, boundary cases, and an optional nightly run with configurable case counts
- Tests compare returned currents, tick progression, queued delayed spikes, and complete checkpoint state after restore and every subsequent event
- Restore compatibility is checked through JSON and postcard, including spikes still in flight, signed weights, empty ticks, delay limits, and multiple deliveries in one slot
- Failing generated cases are reduced to smaller reproducible scenarios with detailed state differences for debugging
- Replaced the unmaintained bincode test dependency with postcard

### Impact
`✅ Detects checkpoint resume mismatches before release`
`✅ Covers delayed spikes during restore`
`✅ Produces smaller, reproducible failures`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

### CODE REVIEW

File: tests/checkpoint_resume/compare.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+pub(crate) fn meshes_equivalent(
+    live: &SynapticMesh,
+    restored: &SynapticMesh,
+) -> Result<(), String> {
+    if live.tick() != restored.tick() {
+        return Err(format!(
+            "tick mismatch: live={} restored={}",
+            live.tick(),
+            restored.tick()
```
[the remark this answers is not in the document]
Comment (rmems) [reply tests/checkpoint_resume/compare.rs]: Addressed in cd501af (Antigravity / Gemini Code Assist): Folded tick check into snapshot comparison and error reporting.

File: tests/checkpoint_resume/harness.rs [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+        Self::DelayCapacity,
+        Self::MultiSpikeSameSlot,
+        Self::CheckpointBeforeDelivery,
+        Self::SignedWeights,
+        Self::EmptyTicks,
+        Self::GeneratedRandom,
+        Self::GeneratedSmallWorld,
+    ];
+
+    pub(crate) fn from_seed(seed: u64) -> Self {
```
[the remark this answers is not in the document]
Comment (rmems) [reply tests/checkpoint_resume/harness.rs:53]: Addressed in cd501af (Antigravity / Gemini Code Assist): Evaluated seed modulo as u64 before indexing: Self::ALL[(seed % Self::ALL.len() as u64) as usize].

File: tests/checkpoint_resume/compare.rs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                }
+            }
+            for k in r_obj.keys() {
+                if !l_obj.contains_key(k) {
+                    diffs.push(format!("  field '{k}' missing in live"));
+                }
+            }
+        }
+        let diff_summary = if diffs.is_empty() {
+            String::new()
```
[the remark this answers is not in the document]
Comment (rmems) [reply tests/checkpoint_resume/compare.rs]: Addressed in b511166 (Antigravity / Gemini Code Assist): Replaced String::new() with String::default().
